css-shared-element-transitions/index.bs

```diff
@@ -593,254 +655,307 @@ document's DOM structure.
-	
-	Issue: We should retarget the animation if computed properties for
-		incoming elements change.
+    To <dfn>animate a page transition</dfn> given a {{SameDocumentTransition}} |transition|:
+
+    1. Generate a <<keyframe>> named "page-transition-fade-out" in
+        [=user-agent origin=] as follows:
+
+        <pre><code highlight=css>
+            @keyframes page-transition-fade-out {
+                  to { opacity: 0; }
+            }
+        </code></pre>
+
+    1. Generate a <<keyframe>> named "page-transition-fade-in" in
+        [=user-agent origin=] as follows:
+
+        <pre><code highlight=css>
+            @keyframes page-transition-fade-in {
+                  from { opacity: 0; }
+            }
+        </code></pre>
+
+    1. Apply the following styles in [=user-agent origin=]:
+
+        <pre><code highlight=css>
+            html::page-transition-outgoing-image(*) {
+                animation: page-transition-fade-out 0.25s both;
+            }
+
+            html::page-transition-incoming-image(*) {
+                animation: page-transition-fade-in 0.25s both;
+            }
+        </code></pre>
+
+    1. [=map/For each=] |tag| -> |capturedElement| of |transition|'s [=SameDocumentTransition/tagged elements=]:
+
+        1. If neither of |capturedElement|'s [=captured element/outgoing image=] or [=captured element/incoming element=] is null:
+
+            1. Let 'transform' be |capturedElement|'s [=outgoing styles=]'s 'transform' property.
+
+            1. Let 'width' be |capturedElement|'s [=outgoing styles=]'s 'width' property.
+
+            1. Let 'height' be |capturedElement|'s [=outgoing styles=]'s 'height' property.
+
+            1. Generate a <<keyframe>> named "page-transition-container-anim-|tag|" in
+                [=user-agent origin=] as follows:
+
+            <pre><code highlight=css>
+                @keyframes page-transition-container-anim-|tag| {
+                    from {
+                        transform: |transform|;
+                        width: |width|;
+                        height: |height|;
+                    }
+                }
+            </code></pre>
+
+        1. Apply the following styles in [=user-agent origin=]:
+
+            <pre><code highlight=css>
+                html::page-transition-container(|tag|) {
+                    animation: page-transition-container-anim-|tag| 0.25s both;
+                }
+            </code></pre>
+
+    1. Let |document| be |transition|'s [=relevant global object's=] [=associated document=].
+
+    1. [=Assert=]: |document|'s [=document/running same document transition=] is null.
+
+    1. Set |document|'s [=document/running same document transition=] to |transition|.
+
+    Issue: How are keyframes scoped to user-agent origin? We could decide
+        scope based on whether `animation-name` in the computed style
+        came from a developer or UA stylesheet.
+        But we do want developers to be able to
```

Comment:
Seems like there was a point midway here?

Reply:
lol I have no idea. I'll delete that line.